### PR TITLE
[DROOLS-6430] Nested map access in RHS fails in exec-model

### PR DESCRIPTION
**Pull request by tkobayas**
**JIRA**: 
https://issues.redhat.com/browse/DROOLS-6430

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by tkobayas** [comment]
Jenkins retest this

**Comment by tkobayas** [comment]
Jenkins retest this

File: drools-model/drools-mvel-compiler/src/main/java/org/drools/mvelcompiler/ast/ListAccessExprT.java [hunk 1]
```diff
@@ -41,6 +44,21 @@ public ListAccessExprT(TypedExpression name, Expression index, Type type) {
...
+        return getType().filter(ParameterizedType.class::isInstance)
+                        .map(ParameterizedType.class::cast)
+                        .map(parameterizedType -> {
+                            Class<?> rawType = (Class<?>) parameterizedType.getRawType();
+                            if (List.class.isAssignableFrom(rawType)) {
+                                return parameterizedType.getActualTypeArguments()[0];
+                            } else if (Map.class.isAssignableFrom(rawType)) {
+                                return parameterizedType.getActualTypeArguments()[1];
+                            } else {
+                                return null;
```
Comment (lucamolteni) [inline drools-model/drools-mvel-compiler/src/main/java/org/drools/mvelcompiler/ast/ListAccessExprT.java:57]: are you sure about this null? I think you should do `flatMap -> Optional`

**Review by lucamolteni — CHANGES_REQUESTED** [review]

File: drools-model/drools-mvel-compiler/src/main/java/org/drools/mvelcompiler/ast/ListAccessExprT.java (hunk 1 above)
Comment (lucamolteni) [reply drools-model/drools-mvel-compiler/src/main/java/org/drools/mvelcompiler/ast/ListAccessExprT.java:57]: I stand corrected that is useless

**Comment by tkobayas** [comment]
Jenkins run fdb

**Review by lucamolteni — APPROVED** [review]

**Review by hellowdan — APPROVED** [review]

**Review by mariofusco — APPROVED** [review]

**Pull request merged**